autovalet_gazebo/scripts/validate_arucotags.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# license removed for brevity
+
+# Author: Nikita Mishra
+# Adapted from: validate_localization.py
+# Date of creation: September 25, 2020 04:20 PM
+# Description: Validation script for Progress Review 8, compare arucotags pose (estimate) with gt.pose
+
+
+import rospy
+from gazebo_msgs.msg import LinkStates
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import PoseStamped
+from tf.transformations import euler_from_quaternion
+from geometry_msgs.msg import Pose
+
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+
+class aruco_tags_validation:
+    def __init__(self, gt_topic, est_topic): #, sample_dist, file):
+        self.gt_handle = rospy.Subscriber(gt_topic, LinkStates, self.gt_cb)
+        self.est_handle = rospy.Subscriber(est_topic, PoseStamped, self.est_cb) ## Odometry --> PoseStamped
+        # self.est_handle = rospy.Subscriber(est_topic, Pose, self.est_cb)
+        # self.sample_dist = sample_dist
+
+        self.gt = None
+        self.estimate = None
+        # self.last_gt = None
+        # self.filename = file
+
+        """
+        To ping the first message of gt and populate the world to pose offset
+        """
+        # self.isFirstCb = True
+        self.offset = None
+        self.positionErrorLog = []
+        self.yawErrorLog = []
+        self.runningAverage = []
+
+    def gt_cb(self, gt):
```

Comment:
more verbose fxn names? `ground_truth_callback`?

Reply:
Let this be the way it is, the function description makes it clearer. 
I kept it this way so that it stays consistent with the other validation scripts.

Cleaned up the rest